jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -2255,18 +2261,39 @@ public boolean hasFullPrivilegeAccess() {
 			STRONG
 		}
 
-		static class ClassDefiner {
-			Class<?> defineClass(boolean option) {
-				return null;
+		static final class ClassDefiner {
+			private final byte[] classBytes;
+			private final String className;
+			private final Lookup lookup;
+			ClassDefiner(String name, byte[] template, Lookup lookupObj) {
+				className = name;
+				classBytes = template;
+				lookup = lookupObj;
+			}
+			Class<?> defineClass(boolean initOption) {
+				JavaLangAccess jlAccess = SharedSecrets.getJavaLangAccess();
+				Class<?> lookupClass = lookup.lookupClass();
+				return jlAccess.defineClass(lookupClass.getClassLoader(), lookupClass, className, classBytes, jlAccess.protectionDomain(lookupClass), initOption, 0, null);
 			}
 		}
 
 		public Lookup defineHiddenClass(byte[] bytes, boolean initOption, ClassOption... classOptions) {
-			return null;
+			ClassReader cr;
+			try {
+				cr = new ClassReader(bytes);
+			} catch (ArrayIndexOutOfBoundsException e) {
+				/*[MSG "K065Y2", "The class byte array is corrupted"]*/
```

Comment:
This should be defined in jcl/src/java.base/share/classes/com/ibm/oti/util/ExternalMessages-MasterIndex.properties.

Reply:
It is already there: 
https://github.com/eclipse/openj9/blob/master/jcl/src/java.base/share/classes/com/ibm/oti/util/ExternalMessages-MasterIndex.properties#L407